### PR TITLE
Ncl 7071 kafka consumer and db source

### PR DESCRIPTION
**Pull request by matejonnet**
@thescouser89 please review the pom changes

### CODE REVIEW

**Comment by thescouser89** [comment]
Not sure what to say but the pom changes look fine and what I'd expect from the Quarkus version jump.

**Review by thescouser89 — APPROVED** [review]

File: src/main/resources/application.yaml [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+  hosts: https://host:443
+  indexes: test-*
+  #keyStorePath =
+  #keyStorePass =
+  #keyPass =
+bifrost:
+  backoffrunnable:
+    delayMillis: 1000
+    maxBackOffCycles: 10
+    pollIntervalMillis: 250
```
Comment (vibe13) [inline src/main/resources/application.yaml:62]: Does this mean that queries will be done every 250ms? In prod, this needs to be increased, as the query might take up to 1 second to execute

File: src/main/java/org/jboss/pnc/bifrost/source/db/LogEntry.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+        name = "LogEntry.findExisting",
+        query = "FROM LogEntry e" + " WHERE e.processContext = :processContext"
+                + "   AND e.processContextVariant = :processContextVariant"
+                + "   AND e.requestContext = :requestContext" + "   AND e.temporary = :temporary"
+                + "   AND e.buildId = :buildId",
+        hints = @QueryHint(name = "org.hibernate.cacheable", value = "true"))
+@Cacheable
+public class LogEntry extends PanacheEntityBase {
+
+    @Id
```
Comment (vibe13) [inline src/main/java/org/jboss/pnc/bifrost/source/db/LogEntry.java:65]: Can you please use a sequence generator:
    @Id
    @SequenceGenerator(
            name = "logEntrySequence",
            sequenceName = "logentry_id_sequence",
            allocationSize = 1,
            initialValue = 1000000)
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "logEntrySequence")

P.S. initial value is an estimation, could be refined later

File: src/main/java/org/jboss/pnc/bifrost/source/db/LogLine.java [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+@Table(
+        indexes = {
+                @Index(name = "idx_logline_eventtimestamp", columnList = "eventTimestamp"),
+                @Index(name = "idx_logline_sequence", columnList = "sequence"),
+                @Index(name = "idx_logline_loggerName", columnList = "loggerName") })
+@JsonDeserialize(using = LogLineDeserializer.class)
+@Cacheable
+public class LogLine extends PanacheEntityBase {
+
+    @Id
```
Comment (vibe13) [inline src/main/java/org/jboss/pnc/bifrost/source/db/LogLine.java:56]: Can you please use a sequence generator:

Can you please use a sequence generator:
    @Id
    @SequenceGenerator(
            name = "logLineSequence",
            sequenceName = "logline_id_sequence",
            allocationSize = 1,
            initialValue = 100000000)
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "logLineSequence")

P.S. initial value is an estimation, could be refined later

**Review by vibe13 — CHANGES_REQUESTED** [review]
Hi @matejonnet  I left a few comments but overall it looks good, massive work! We still need to implement the extract from ES and save in DB right, or have I missed it from below? Thanks!

File: src/main/java/org/jboss/pnc/bifrost/source/db/LogEntry.java (hunk 2 above)
Comment (matejonnet) [reply src/main/java/org/jboss/pnc/bifrost/source/db/LogEntry.java:65]: It's intentional that the ID is generated in the application as I believe generating the "globally" unique ID (same as in other PNC services) in the application performs better:
- no need for another IO operation to get the ID from the DB
- no need to sync the sequence in the DB cluster (geo disrtibution)

**Comment by matejonnet** [comment]
> Hi @matejonnet I left a few comments but overall it looks good, massive work! We still need to implement the extract from ES and save in DB right, or have I missed it from below? Thanks!

The "extract from ES" is a one time job, as we discussed it's probably the easiest way to do it with Bifrost, but I would not include it in the release, we can do it in a branch that we run "standalone".

File: src/main/java/org/jboss/pnc/bifrost/endpoint/provider/DataProvider.java [hunk 4]
```diff
@@ -110,13 +123,20 @@ public void subscribe(
         subscriptions.subscribe(subscription, searchTask, afterLine, onLine, backOffRunnableConfig);
     }
 
+    /**
+     * Example filters:
+     *
+     * "prefixFilters": loggerName.keyword: org.jboss.pnc.causeway|org.jboss.pnc._userlog_,
+     * level.keyword:INFO|ERROR|WARN, "matchFilters": mdc.buildId.keyword:ARTPKTKTQGAAC,
```
[the remark this answers is not in the document]
Comment (vibe13) [reply src/main/java/org/jboss/pnc/bifrost/endpoint/provider/DataProvider.java]: @matejonnet the `buildPushResultId` is global now, but for old records (which used an autoincrement / sequence) there are clashes with buildrecord.id (I found 4414 in orch prod DB)

File: src/main/resources/application.yaml (hunk 1 above)
Comment (vibe13) [reply src/main/resources/application.yaml:62]: ah ok, sounds good then. I would anyway go for 500 / 750, but if there are no concurrent executions, it's fine. thanks!

File: src/main/java/org/jboss/pnc/bifrost/source/db/LogEntry.java (hunk 2 above)
Comment (vibe13) [reply src/main/java/org/jboss/pnc/bifrost/source/db/LogEntry.java:65]: ah ok, makes sense, I was afraid it was going to be an autoincrement id. ok so the logentry.id should be mapped as bigint also.

**Review by vibe13 — APPROVED** [review]

**Pull request merged**